csharp/src/Ice/InputStream.cs

```diff
@@ -2657,7 +2651,7 @@ public PatchEntry(System.Action<Value> cb, int classGraphDepth)
         };
 
         internal EncapsDecoder(InputStream stream, Encaps encaps, bool sliceValues,
-                               int classGraphDepthMax, ValueFactoryManager f)
+                               int classGraphDepthMax, ValueFactoryManager? f)
```

Comment:
When do we never have a value factory manager? If there's no communicator?

Reply:
That's indeed incorrect. We can't unmarshal a class without a communicator since instance provides the activator.

Reply:
Actually it's correct. We can use this EncapsDecoder for decoding other types with a null communicator / value factory manager. Fixed other uses.